code/modules/clothing/suits/miscellaneous.dm

```diff
@@ -958,7 +958,9 @@
 	if(linked_staff.faith >= 100)	//if the linked staff is fully recharged, do nothing
 		return
 
-	if(!(linked_staff in range(3, get_turf(src))))		//staff won't charge at range (to prevent it from being handed off / stolen and used)
+	// Do not allow the staff to recharge if it's more than 3 tiles away from the robe. If get_dist returns 0, the robe and the staff in the same tile.
+	// Using .loc instead of get_turf() here because if the turf of H is the same as the turf of the staff, get_dist() returns -1.
+	if(!(get_dist(H.loc, linked_staff.loc) in 0 to 3))
```

Comment:
Can write this easier `if(get_dist(get_turf(H), get_turf(linked_staf)) > 3)`
Can most likely also be done without the `get_turf` and instead with the `loc` or just by using `H` and `linked_staff` but you'll have to check that.

Reply:
Yeah I had something like that before I settled on this. There needs to be a 0 case for when the two items are on the same tile. But, as said in my comment there, if you use `get_turf` and both `get_turf()`s return the same turf, `get_dist` gives you a -1, which makes the check fail. 

However, in actuality, that's valid for the staff to be recharging. Also, the refs for get_dist said it can return -1 for other error conditions, so that's why I didn't want to mess with using get_turf.

Reply:
Hmm looking at the reference. It doesn't specify other cases than the object being the same. Maybe it also returns `-1` when they are on other Z levels? If so that might be problematic

Having the `in 0 to 3` here is the thing that I find odd so to say. Most checks use a simple `get_dist > value`.

Reply:
Yeah I want to see if it would work with just 1 get_dist call, I know its probably weird :smile:
I could do something like `if(get_dist(H, staff) > 3 || get_dist(H, staff) < 0)` instead I suppose.